### ISSUE TITLE
Trying to find file load and path resolution logic

### ISSUE THREAD

**Issue by runvnc** [mamedev/mame#2268]
TLDR: any help finding which code files and functions determine the path and try to load files with this command line: `mame64.exe coco3 -cart dungeonsofdaggorath.ccc`

Hello, amazing software here.  I have been having some issues loading MESS-type things like coco3 etc. and would like to clarify (by looking at the loading code) exactly what paths I should be placing floppies/tapes/cart images and BIOS files in.  I am actually thinking of changing my fork to make this easier/more forgiving somehow.. if I can figure out how it works.

So I have actually spent many hours trying to insert logging messages etc. to pin down exactly how it works.  I have a number of candidate locations in the code that seem important, but my attempts to insert debug information by logging have left me quite uncertain how it is determining the paths and file loading code.

Right now, it seems that I can load a coco3 cart via mame64.exe coco3 -cart dungeonsofdaggorath.ccc only if the ccc file is in the directory with the executable.  I have found emu/romload.cpp to be quite interesting and a strong possibility.  Unfortunately when I do not place the ccc file in the directory with the executable, it does not seem to enter the handle_missing_file function like I hoped.  Instead it seems to run into some divide by zero error in osd/eminline.

I am just trying to track down the process it is using to determine the paths to look in and to try to load the files.

Thank you very much for any hints you can give me.

**Comment by cuavas** [comment]
ROM/software path resolution is only used when you're loading things from built-in system definitions (e.g. arcade games or computer BIOS ROMs) or from software lists (stuff defined in XML files from the hash directory).  Loose software software requires you to provide the full path to the image file.

**Comment by runvnc** [comment]
@cuavas thank you, that is an excellent hint.  In my fork I would like to change it so that rather than a full path, it can look in some common paths or paths based on the type of system.  Can you give me any idea which code file(s) I should  modify to change this?

**Comment by Robbbert** [comment]
The beginning path for loose software is set by the swpath setting in the game's ini file. You can have an ini for a source-file, thus fullfilling your requirement.

If you have further questions, please ask on the forum.
